app/models/repository.rb

```diff
@@ -405,4 +408,39 @@ def hide
   def gather_maintenance_stats_async(priority: :medium)
     RepositoryMaintenanceStatWorker.enqueue(id, priority: priority)
   end
+
+  # Set the unmaintained status for this Repository and related Projects if the readme file
+  # indicates that the repository is not currently being maintained. This method will also
+  # removed the unmaintained status if neither the readme or repository indicate that it
+  # is not being maintained in the case where a project was picked up for work again.
+  def update_unmaintained_status_from_readme
+    if readme.unmaintained?
+      # update repository status only if needed
+      update(status: "Unmaintained") unless unmaintained?
```

Comment:
the `unless unmaintained?` check might be unnecessary because ActiveModel::Dirty will skip saving if no values have changed, but also can't harm to keep it here.